tests/test_api.py

```diff
@@ -473,3 +473,110 @@ def test_unenrolls_api(self):
             data['population']['variant'][1],
             {'window': self.window2[0].isoformat(), 'count': 10}
         )
+
+
+class EnrollmentBaseTestCase(TestCase):
+
+    def create_data(self):
+        self.window_start = datetime.datetime(2018, 1, 4, 16, 0, 0, tzinfo=pytz.UTC)
+        # This will create cases where window_start > window_end below which should be OK.
+        self.window_end = datetime.datetime(2018, 1, 4, 16, 1, 1, tzinfo=pytz.UTC)
+
+        Enrollment.objects.create(
+            type='preference_study',
+            experiment='pref-flip-1',
+            branch='control',
+            window_start=self.window_start,
+            window_end=self.window_end,
+            enroll_count=10,
+            unenroll_count=1
+        )
+        Enrollment.objects.create(
+            type='preference_study',
+            experiment='pref-flip-1',
+            branch='control',
+            window_start=self.window_start + datetime.timedelta(minutes=5),
+            window_end=self.window_end,
+            enroll_count=20,
+            unenroll_count=2
+        )
+        Enrollment.objects.create(
+            type='preference_study',
+            experiment='pref-flip-1',
+            branch='variant',
+            window_start=self.window_start + datetime.timedelta(minutes=10),
+            window_end=self.window_end,
+            enroll_count=11,
+            unenroll_count=5
+        )
+        Enrollment.objects.create(
+            type='preference_study',
+            experiment='pref-flip-1',
+            branch='variant',
+            window_start=self.window_start + datetime.timedelta(minutes=15),
+            window_end=self.window_end,
+            enroll_count=21,
+            unenroll_count=10
+        )
+        # Create a random bit of data in a different experiment.
+        Enrollment.objects.create(
+            type='preference_study',
+            experiment='pref-flip-2',
+            branch='control',
+            window_start=self.window_start + datetime.timedelta(minutes=5),
+            window_end=self.window_end,
+            enroll_count=99999,
+            unenroll_count=99
+        )
```

Comment:
My recommendation: Scrap this base class. Update the existing base class to change the `timezone.now()` to be a fixed datetime just after a day boundary (so 12:05am). We should add a few extra `Enrollment` records as well. By doing so we can use the base class for both realtime and all time counts since the daily boundary will test the all time counts by testing the aggregation per day.

Reply:
I have `now` replaced with:

base_time = datetime.datetime(2018, 1, 4, 16, 50, 0, tzinfo=pytz.UTC)
self.window1 = (base_time, base_time + datetime.timedelta(minutes=5))
self.window2 = (base_time + datetime.timedelta(minutes=5), base_time + datetime.timedelta(minutes=10))

This results in:

File "/app/tests/test_api.py", line 351, in test_population_api
    data['population']['control'][0],
IndexError: list index out of range

I can't seem to get further than this (these are existing tests - though I get the same on the ones I tried to write). Not sure how to diagnose this, even though the error message is clear, since I don't know how the test suite actually sees the data.